pkgs/misc/emulators/citra/base.nix

```diff
@@ -0,0 +1,215 @@
+{ pname, version, src, branchDesc, branchName
+, lib, stdenv, config, fetchFromGitHub
+, catch2, cmake, ninja, pkg-config, wrapQtAppsHook
+, enet, fmt, inih, libressl, soundtouch, zstd, discord-rpc
+, boost173, ffmpeg, libusb1, qtbase, qtdeclarative, qtmultimedia, qttools, rapidjson, SDL2
+, alsaSupport      ? stdenv.isLinux,                      alsa-lib      ? null
+, discordSupport   ? true
+, fdkSupport       ? true,                                fdk_aac       ? null
+, jackaudioSupport ? true,                                libjack2      ? null
+, onlineSupport    ? true
+, pulseSupport     ? config.pulseaudio or stdenv.isLinux, libpulseaudio ? null
+, sndioSupport     ? true,                                sndio         ? null
+, udevSupport      ? stdenv.isLinux,                      udev          ? null
+}:
+
+with lib;
+let
+  cpp-jwt-src = fetchFromGitHub {
```

Comment:
If we're gonna devendor all submodules like this, I think we should have an update script for them. Otherwise updating the packages becomes a real pain, having to manually go through them all. I don't think that's a good idea.

Reply:
Indeed, would you be willing to work on such a script? You seem to have experience in that area. I'd also like to spin some of these off into their own derivations when it makes sense.

Reply:
Indeed, would you be willing to work on such a script? You seem to have experience in that area. I'd also like to spin some of these off into their own derivations when it makes sense.

Reply:
I have written a small script which fetches the submodules, and exports the data `fetchFromGitHub` needs as a json file. The idea is to use `lib.importJSON` on that, and extract the result to the location citra expects. It was fun to write, but I think it's just not practical to maintain. The script is currently very unfinished, as which it will probably stay. Here's the current (very messy) state: https://github.com/IvarWithoutBones/nixpkgs/commit/4e155ef7fee11283d5dd15ae975f615c01f51bc4.

I think a better option would be working with upstream, to properly devendor these. Dynarmic is an dependency of Citra and yuzu (currently a submodule) which takes ages to compile. It recently added an cmake install target\[[1](https://github.com/merryhime/dynarmic/pull/652)\], adding support for that in upstream yuzu and citra would make testing much faster. Right now maintaining yuzu is kind of a pain for me, since I don't have a very fast computer.

I hope adding support for it shouldn't be that difficult, though I'm not extremely familair with CMake.

Reply:
I'm switching back to fetchSubmodules, we should indeed investigate working with upstream and devendoring in the future.